### DISCUSSION TITLE
Sending emails with new line characters  "\n"  in the body text

### DISCUSSION THREAD

**Discussion by MaskellRichard** [O365/python-o365#930] [Q&A]
I'm trying to send an email where the body text contains new line "\n" characters. The code that sets the body text is:

msg.body = requestString + "\n" + response

Whatever I do the emails body text has had the new line character replaced with a space character. I've confirmed this by copying the body text from the generated email (sent to my inbox) and pasting it into an appropriate development tool and viewing invisible characters.

Please don't tell me I don't need the newline character, it's not possible to change this. 

The code reads a shared Office 365 Inbox, analyses the emails and sends out emails in response. The code has run for several years using EasyIMAP to read the emails from a shared Office365 mailbox. Microsoft recently breaking/withdrawing support of IMAP has prompted the migration to O365. Reading the emails with O365 is working fine. The old code uses SMTPLib to send the emails. I can stick with this so my need isn't urgent. However I don't trust Microsoft not to withdraw support for SMTP access at some time in the future and would rather fix this sooner than later.

**Comment by alejcas** [comment]
This is what body setter does:

```python
@body.setter
    def body(self, value):
        if self.__body:
            if not value:
                self.__body = ''
            elif self.body_type == 'html':
                soup = bs(self.__body, 'html.parser')
                soup.body.insert(0, bs(value, 'html.parser'))
                self.__body = str(soup)
            else:
                self.__body = ''.join((value, '\n', self.__body))
        else:
            self.__body = value
        self._track_changes.add('body')
```

As you see when it's html it's inserting the vale at position 0.
When it's text its already adding '\n'

**Reply by MaskellRichard** [reply]
Fantastic. That's fixed it. 

In my code I used

                msg.body_type = "Text"
              
Initially I tried:

                msg.body_type = ""
                
However this threw an error due to body_type having a Null value.

When I looked at this before posting on this Forum I'd found documentation on how to extract the body text from a message as plain text rather than HTML. I couldn't find any examples of sending a message where the body type was anything other than HTML and wasn't bright enough to work out the syntax.

Thanks again
